core/src/main/scala/io/qbeast/core/transform/LinearTransformer.scala

```diff
@@ -32,14 +32,21 @@ case class LinearTransformer(columnName: String, dataType: QDataType) extends Tr
       statsSqlPredicates = Seq(s"max($columnName) AS $colMax", s"min($columnName) AS $colMin"))
 
   override def makeTransformation(row: String => Any): Transformation = {
-    val min = getValue(row(colMin))
-    val max = getValue(row(colMax))
-    assert(min != null && max != null)
-    dataType match {
-      case ordered: OrderedDataType =>
-        LinearTransformation(min, max, ordered)
-
+    val minAux = row(colMin)
+    val maxAux = row(colMax)
+    if (minAux == maxAux) {
+      // If all values are null or equal we return an IdentityTransformation
+      IdentityTransformation
```

Comment:
Maybe is best just to put a new NullToZeroTransformation that does only that, instead of adding the case null => 0.0 in the identity. It is more explicitly. And also, it is not identity

Reply:
Cool!